aptos-move/block-executor/src/executor.rs

```diff
@@ -603,31 +638,34 @@ where
         let init_timer = VM_INIT_SECONDS.start_timer();
         let executor = E::init(executor_arguments);
         drop(init_timer);
+
         let data_map = UnsyncMap::new();
+        let mut unsync_view = LatestView::<T, S, X>::new_unsync_view(base_view, &data_map);
 
         let mut ret = Vec::with_capacity(num_txns);
 
         let mut accumulated_fee_statement = FeeStatement::zero();
 
         for (idx, txn) in signature_verified_block.iter().enumerate() {
-            let res = executor.execute_transaction(
-                &LatestView::<T, S, X>::new_btree_view(base_view, &data_map, idx as TxnIndex),
-                txn,
-                idx as TxnIndex,
-                true,
-            );
+            unsync_view.update_txn_idx(idx as TxnIndex);
+            let res = executor.execute_transaction(&unsync_view, txn, idx as TxnIndex, true);
 
             let must_skip = matches!(res, ExecutionStatus::SkipRest(_));
             match res {
                 ExecutionStatus::Success(output) | ExecutionStatus::SkipRest(output) => {
                     assert_eq!(
-                        output.get_deltas().len(),
+                        output.aggregator_v1_delta_set().len(),
                         0,
                         "Sequential execution must materialize deltas"
                     );
                     // Apply the writes.
-                    for (ap, write_op) in output.get_writes().into_iter() {
-                        data_map.write(ap, write_op);
+                    for (key, write_op) in output
+                        .resource_write_set()
```

Comment:
I didn't expose the trait for that, but I could, however, it would also have modules, right? I was thinking since we will get rid of v1, no need to have writes + v1 as an API outside.

Reply:
Ahh, true. Having this is better :)